### ISSUE TITLE
Inconsistent behaviour of init containers when Pod has invalid security profiles (apparmor vs seccomp)

### ISSUE THREAD

**Issue by pjbgf** [kubernetes/kubernetes#97273]
**What happened**:

Pods with invalid container-level seccomp profiles allow its init containers to be executed. Pods with unloaded apparmor profiles blocks the entire pod.

**What you expected to happen**:

Consistency between both. In one hand AppArmor blocks all pod execution regardless of the containers having or not a profile attached to it. On the other, Seccomp allows for an init container to deploy its own profile - as currently done on [kubernetes-sigs/security-profiles-operator](https://github.com/kubernetes-sigs/security-profiles-operator).

**How to reproduce it (as minimally and precisely as possible)**:

Apply the yaml below:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: apparmor-sample
  namespace: default
  annotations:
    container.apparmor.security.beta.kubernetes.io/container-with-apparmor: localhost/non-existent-profile
spec:
  initContainers:
    - name: init-test
      image: alpine
      command: [echo, "the init container was executed!"]
  containers:
    - name: container-without-apparmor
      image: alpine
    - name: container-with-apparmor
      image: alpine
---
apiVersion: v1
kind: Pod
metadata:
  name: seccomp-sample
  namespace: default
spec:
  initContainers:
    - name: init-test
      image: alpine
      command: [echo, "the init container was executed!"]
  containers:
    - name: container-without-seccomp
      image: alpine
    - name: container-with-seccomp
      image: alpine
      securityContext:
        seccompProfile:
          type: Localhost
          localhostProfile: non-existent-profile.json
```

Resulting in only the init container from the `seccomp-sample` being executed:
![image](https://user-images.githubusercontent.com/5452977/102027847-c0943e80-3d9e-11eb-95a4-be3bb3d9198a.png)

**Anything else we need to know?**:

AppArmor is currently being target to GA on 1.21 (https://github.com/kubernetes/enhancements/pull/1444). 

**Environment**:
- Kubernetes version (use `kubectl version`):
`Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:59:43Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.0-alpha.0.292+87984d84d18c78", GitCommit:"87984d84d18c7816835e5efedebb17ed13aa2509", GitTreeState:"clean", BuildDate:"2020-12-12T23:13:21Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}`
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): `Ubuntu 18.04.5 LTS`
- Kernel (e.g. `uname -a`):
`Linux XXXXXX 4.15.0-128-generic #131-Ubuntu SMP Wed Dec 9 06:57:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/sig node
/area security

**Comment by saschagrunert** [comment]
I think we can consider this as a bug, right? Then we could work on a fix for v1.21 interdependently of the AppArmor GA graduation.

**Comment by pjbgf** [comment]
Sounds like a good plan @saschagrunert. :+1:
